rootfs/etc/profile.d/_50-workspace.sh

```diff
@@ -0,0 +1,31 @@
+# Files in the profile.d directory are executed by the lexicographical order of their file names.
+# This file is named _50-workspace.sh. The leading underscore is needed to ensure this file
+# executes before other files that may depend on it.
+# The number portion is to ensure proper ordering among the high-priority scripts.
+# This file depends on colors.sh, localhost.sh, and preferences.sh and must come after them
+#
+
+# file_on_host is true when the argument is a file or directory that appears to be on the Host file system.
+
+function file_on_host() {
+	local file
+	file="$(readlink -e "$1")" || return 1
+	local path
+	for path in "${GEODESIC_HOST_PATHS[@]}"; do
+		# Skip paths that are just slashes, or completely empty, which would match everything
+		[[ "$path" =~ ^/*$ ]] && continue
+		if [[ "$path" == "${file}/" || "${file}" == "$path"* ]]; then
+			return 0
+		fi
+	done
+	return 1
+}
```

Comment:
The regex `^/*$` matches the empty path, no need for a separate `-z` test. 

I like the name "file" because it represents the target of "file_on_host".